src/native/minipal/utf8.h

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_UTF8_H
+#define HAVE_MINIPAL_UTF8_H
+
+#include <minipal/utils.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+#define MB_ERR_INVALID_CHARS 0x00000008
+#define ERROR_NO_UNICODE_TRANSLATION 1113L
+#define ERROR_INSUFFICIENT_BUFFER 122L
+#define ERROR_INVALID_PARAMETER 87L
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif // __cplusplus
+
+#ifdef TARGET_WINDOWS
+typedef wchar_t CHAR16_T;
+#else
+typedef unsigned short CHAR16_T;
```

Comment:
Can we use `char16_t`? Or is that no available in the C version that we use?

Reply:
`char16_t` is `uint16_t` in C11, but we are using C99 so unsigned short is used. I had to uppercase `char16_t` typedef for Windows.

Reply:
`char16_t` from C++11 is `uint16_t` in C11, but we are using C99 so I've used `unsigned short`. I had to uppercase `char16_t` typedef for Windows.

Reply:
`char16_t` from C++11  / C11 is defined as `uint16_t`, but we are using C99 so I've used `unsigned short`. I had to uppercase `char16_t` typedef because it was conflicting on Windows.

Reply:
I tried that as well, but CI build on some platform was failing (can’t remember the exact leg).